### PR TITLE
Add missing docs for the rest of the util module

### PR DESCRIPTION
**Pull request by qinsoon**
This PR is a step towards https://github.com/mmtk/mmtk-core/issues/309.
* Deny `missing_docs` for the `util` module and the `vm` module.
* Change some items from `pub` to `pub(crate)`.
* Remove some unused constants.

### CODE REVIEW

**Comment by qinsoon** [comment]
binding-refs
OPENJDK_BINDING_REPO=qinsoon/mmtk-openjdk
OPENJDK_BINDING_REF=update-mmtk-core-pr-1026

File: src/util/alloc/bumpallocator.rs [hunk 1]
```diff
@@ -32,11 +34,14 @@ pub struct BumpAllocator<VM: VMBinding> {
 #[repr(C)]
 #[derive(Copy, Clone)]
 pub struct BumpPointer {
+    /// The current address of the allocation buffer. This is what will be allocated next.
```
Comment (wks) [inline src/util/alloc/bumpallocator.rs:37]: I would not say the `cursor` *is* the current address of the allocation buffer itself.  Intuitively, the buffer doesn't move, but the cursor moves.  The buffer may be a whole block or a consecutive range of lines in an Immix block, but it is chosen in the slow path as the range from the *initial* `cursor` to the `limit`.  We don't keep track of the starting point of the buffer, but conceptually the cursor is not the "address of the allocation buffer".

```suggestion
    /// The cursor inside the allocation buffer where the next object will be allocated.
```

File: src/util/conversions.rs [hunk 2]
```diff
@@ -70,6 +78,7 @@ pub fn bytes_to_pages_up(bytes: usize) -> usize {
     raw_align_up(bytes, BYTES_IN_PAGE) >> LOG_BYTES_IN_PAGE
 }
 
+/// Convert size in bytes to the number of pages (aligned down)
```
Comment (wks) [inline src/util/conversions.rs:81]: The doc comment says the result is aligned *down*, but it calls `bytes_to_pages_up`.  Is this a mistake?

File: src/util/conversions.rs [hunk 3]
```diff
@@ -2,65 +2,73 @@ use crate::util::constants::*;
...
     unsafe { Address::from_usize(chunk << LOG_BYTES_IN_CHUNK) }
 }
 
+/// Align up an integer to the given alignment.
 pub const fn raw_align_up(val: usize, align: usize) -> usize {
     // See https://github.com/rust-lang/rust/blob/e620d0f337d0643c757bab791fc7d88d63217704/src/libcore/alloc.rs#L192
     val.wrapping_add(align).wrapping_sub(1) & !align.wrapping_sub(1)
 }
 
+/// Align down an integer to the given alignment.
```
Comment (wks) [inline src/util/conversions.rs:61]: Need to mention that `align` must be a power of two, otherwise it won't work.

File: src/util/conversions.rs [hunk 4]
```diff
@@ -2,65 +2,73 @@ use crate::util::constants::*;
...
 pub const fn raw_align_down(val: usize, align: usize) -> usize {
     val & !align.wrapping_sub(1)
 }
 
+/// Is the integer aligned to the given alignment?
```
Comment (wks) [inline src/util/conversions.rs:66]: Ditto.

File: src/util/mod.rs [hunk 5]
```diff
@@ -39,6 +39,7 @@ pub(crate) mod erase_vm;
 pub(crate) mod finalizable_processor;
 /// Heap implementation, including page resource, mmapper, etc.
 pub mod heap;
+/// Checking if an address is an valid MMTk object.
```
Comment (wks) [inline src/util/mod.rs:42]: I am considering moving those doc comments into their respective modules in the form of `//!` at the beginning of the file so that we can have more detailed documentation for each module.  But we may do it in a separate PR.

**Review by wks — CHANGES_REQUESTED** [review]

File: src/util/conversions.rs (hunk 2 above)
Comment (qinsoon) [reply src/util/conversions.rs]: Yes, it was a mistake when I added the comment. I thought it was 'align down', as we have the other function for aligning up. 

I think it is a bit confusing to have two functions for the same thing but with different names. I removed this function, and use `bytes_to_pages_up` instead.

File: src/util/mod.rs (hunk 5 above)
Comment (qinsoon) [reply src/util/mod.rs:42]: I added the comments in a format that is consistent with the other comments in the same file. If we want to move the docs to the modules, we should systematically do so for the code base. I don't plan to do that in the PR.

**Comment by qinsoon** [comment]
@wks  I have made most of the changes as suggested, and replied to the other comments.

**Review by wks — APPROVED** [review]
LGTM
